src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -365,11 +365,12 @@ mode with existing flows and hold flows of different table-mode types"() {
         List<PathNode> desiredPath = null
         List<Switch> involvedSwitches = null
         def switchPair = topologyHelper.allNotNeighboringSwitchPairs.collectMany { [it, it.reversed] }.find { pair ->
-            desiredPath = pair.paths.find { path ->
-                involvedSwitches = pathHelper.getInvolvedSwitches(path)
-                involvedSwitches.size() == 3 &&
-                        involvedSwitches.every { it.features.contains(SwitchFeature.MULTI_TABLE) }
+            def allPaths = pair.paths.findAll { path ->
+                pathHelper.getInvolvedSwitches(path).every { it.features.contains(SwitchFeature.MULTI_TABLE) }
             }
+            desiredPath = allPaths.find { pathHelper.getInvolvedSwitches(it).size() == 3 }
+            // make sure that alternative path for protected path is available
+            allPaths.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }.size() >= 2
```

Comment:
I don't really follow how desired path is connected with this line. So all paths may have a protected path available, but there are no guarantees that the picked desiredPath will have a suitable protected path

Reply:
can't agree more.
fixed